test/Kestrel.Transport.FunctionalTests/ResponseTests.cs

```diff
@@ -802,7 +803,14 @@ async Task App(HttpContext context)
             Assert.False(requestAborted);
         }
 
+        private bool ConnectionNotClosedWhenClientSatisfiesMinimumDataRateGivenLargeResponseHeadersRetryPredicate(Exception e)
+            => e is IOException && e.Message.Contains("Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request");
```

Comment:
Should this be made static? Can we share predicates?

Reply:
It wouldn't make it that much better. Currently, we look for the predicate in the type that contains the actual type. I could move this to a base class or add a call to a static utility method elsewhere but I'm not sure it's worth it for just two tests.